Documentation/Contributing.md

```diff
@@ -0,0 +1,117 @@
+## Code Contributions
+There are several types of contributions one can make. Bug fixes, documentation and enhancements that do not materially change the user facing semantics of Swift Package Manager should be submitted directly as PR. 
+
+Larger changes that do materially change the semantics of Swift Package Manager (e.g. changes to the manifest format or behavior) are required to go through [Swift Evolution Process](https://github.com/apple/swift-evolution/blob/master/process.md).
+To see how previous evolution decisions for SwiftPM have been made check out https://github.com/apple/swift-package-manager/blob/main/Documentation/Internals/PackageManagerCommunityProposal.md.  
+
+For more information about making contributions to the Swift project in general see [Swift Contribution Guide](https://swift.org/contributing/)  
+
+### Ways to contribute
+Report a bug guide: https://github.com/apple/swift-package-manager/blob/main/Documentation/Resources.md#reporting-a-good-swiftpm-bug.  
+JIRA Bug Tracker (a place where you can open bugs, enhancements to start to contribute): [https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22](https://bugs.swift.org/browse/SR-13640?jql=component%20%3D%20%22Package%20Manager%22).
+
+### Requirements
+You have multiple ways to setup your development environment, here we will focus on two:  
+* [Using Xcode](#using-xcode)
+* [Using a standalone Swift toolchain](#using-standalone)
+
+<a id="using-xcode">*A) _Use Xcode to setup what you need_*.</a>  
+Xcode is only available for macOS.
+
+1. Install latest Xcode - [https://developer.apple.com/xcode](https://developer.apple.com/xcode/)
+2. Confirm you have the latest SwiftPM:
+```
+$> swift package --version
+Swift Package Manager - Swift 5.3.0
+```
+3. Confirm you have the latest Swift version:
+```
+$> swift --version
+Apple Swift version 5.3
+```
+
+<a id="using-standalone">*B) _Use standalone Swift toolchain_*</a>:  
+Download the toolchain from https://swift.org/download/ (or use tools like swiftenv) and confirm its installed correctly like step 2 and 3.
+
+### <a name="getting-started">Getting Started</a>
+1. Pull the SwiftPM repository:
+```
+git clone https://github.com/apple/swift-package-manager.git
+```
+2. Run your first build:
+```
+$> cd swift-package-manager
+$> swift build
+```
+Make sure the build did not fail.  
+
+A `.build/` folder will be generated and it should have inside a similar structure (including build binaries):
+```
+artifacts/
+checkouts/
+debug/
+repositories
+x86_64-apple-macosx
+debug.yaml
+manifest.db
+workspace-state.json 
+```
+Binaries (in the example above) are in `x86_64-apple-macosx/`.  
+If you need to build the generated binaries, run `swift-build` in inside `.build/`:
+```
+./.build/x86_64-apple-macosx/debug/swift-build
+```
+
+### Code Editor
+If you are contributing using macOS, the best option is to use Xcode to build and run test SwiftPM. 
```

Comment:
move this sentence in the "development environment" section and remove the "code editor" title